File: launchdarkly/csharp/src/OSEG.LaunchDarklyExamples/PostRepositoryExample.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text.Json;

using Org.LaunchDarklyTools.Api;
using Org.LaunchDarklyTools.Client;
using Org.LaunchDarklyTools.Model;

namespace OSEG.LaunchDarklyExamples;

public class PostRepositoryExample
{
    public static void Run()
    {
        var config = new Configuration();
        config.ApiKey = new Dictionary<string, string> {["ApiKey"] = "YOUR_API_KEY"};

        var repositoryPost = new RepositoryPost(
            name: "LaunchDarkly-Docs",
            sourceLink: "https://github.com/launchdarkly/LaunchDarkly-Docs",
            commitUrlTemplate: "https://github.com/launchdarkly/LaunchDarkly-Docs/commit/${sha}",
            hunkUrlTemplate: "https://github.com/launchdarkly/LaunchDarkly-Docs/blob/${sha}/${filePath}#L${lineNumber}",
            type: RepositoryPost.TypeEnum.Github,
            defaultBranch: "main"
        );

        try
        {
            var response = new CodeReferencesApi(config).PostRepository(
                repositoryPost: repositoryPost
            );

            Console.WriteLine(response);
        }
        catch (ApiException e)
        {
            Console.WriteLine("Exception when calling CodeReferencesApi#PostRepository: " + e.Message);
            Console.WriteLine("Status Code: " + e.ErrorCode);
            Console.WriteLine(e.StackTrace);
        }
    }
}

File: launchdarkly/typescript-axios/src/PostRepositoryExample.ts
import axios, { AxiosError } from "axios";
import * as api from "launchdarkly_client"

const configuration = new api.Configuration({
  apiKey: "YOUR_API_KEY",
});

const repositoryPost: api.RepositoryPost = {
  name: "LaunchDarkly-Docs",
  sourceLink: "https://github.com/launchdarkly/LaunchDarkly-Docs",
  commitUrlTemplate: "https://github.com/launchdarkly/LaunchDarkly-Docs/commit/${sha}",
  hunkUrlTemplate: "https://github.com/launchdarkly/LaunchDarkly-Docs/blob/${sha}/${filePath}#L${lineNumber}",
  type: api.RepositoryPostTypeEnum.Github,
  defaultBranch: "main",
};

new api.CodeReferencesApi(configuration).postRepository(
  repositoryPost,
).then(response => {
  console.log(response.data);
}).catch((error: Error | AxiosError) => {
  console.log("Exception when calling CodeReferencesApi#postRepository:");
  axios.isAxiosError(error) ? console.log(error.message) : console.log(error);
});
